API/common/exception.py

```diff
@@ -1,25 +1,75 @@
-class ParameterException(Exception):
-    def __init__(self, msg: str) -> None:
-        super().__init__()
-        self.msg = msg
+from common.util import generate_response
 
 
 class CopilotException(Exception):
+    ERROR_CODE = 501
+
     def __init__(self, msg: str) -> None:
         super().__init__()
         self.msg = msg
 
+    def to_response_body(self) -> str:
+        return generate_response([], self.ERROR_CODE, self.msg)
+
+
+class UserException(CopilotException):
```

Comment:
Actually, they are two completely different types of errors, and there is no inheritance relationship, right?